mediacapture-image/MediaStreamTrack-getConstraints-fast.html

```diff
@@ -35,45 +35,31 @@
 
 // These tests verify that MediaStreamTrack.getConstraints() exists and that,
 // returns the constraints passed beforehand with applyConstraints.
-var makeAsyncTest = function(c) {
-  async_test(function(t) {
+function makePromiseTest(name, c) {
+  promise_test(async function() {
     var stream = canvas.captureStream();
     var videoTrack = stream.getVideoTracks()[0];
 
     const constraintsIn = { advanced : [ c ]};
 
-    // Method applyConstraints() will fail since there is no Image Capture
-    // service in this Layout Test, but |constraintsIn| should be cached.
-    videoTrack.applyConstraints(constraintsIn)
-        .then(() => { /* ignore */ })
-        .catch((e) => { /* ignore */ })
-        .then(() => {
-          const constraintsOut = videoTrack.getConstraints();
-          assert_object_equals(constraintsOut, constraintsIn, "constraints");
-          t.done();
-        });
+    await videoTrack.applyConstraints(constraintsIn);
```

Comment:
Note that this changes the behavior of the test. This being rejected is no longer tolerated. I tried wrapping this in try/catch, but even if one suppresses the error the following assert still fails. Feedback appreciated.

Reply:
My recollection is that `async_test` runs in parallel while `promise_test` runs in sequence. The failure may be caused by the original test being buggy. Can you upload the error where you wrap the `applyConstraints()` call in a try/catch to ignore any errors?

Reply:
That's right, `promise_test` run in sequence. The way this was written it should have been fine to race all the subtests with `async_test`, I switched to `promise_test` just to make the test clearer.

Reply:
I've pushed a change with the try/catch to show what happens.

Reply:
I'm sorry but I'm having some trouble navigating the CI results. What is the error causing the failure? If it is an `assert_object_equals()` what are the actual and expected values?

Reply:
After we were embarrassed at TPAC to realize how obscure the "Expand" button is and you learned of its existence, can you tell from the results at which step this failed and if you think this is the right failure mode?